base/compiler/typelimits.jl

```diff
@@ -543,39 +543,40 @@ end
 # compute typeintersect over the extended inference lattice
 # where v is in the extended lattice, and t is a Type
 function tmeet(@nospecialize(v), @nospecialize(t))
-    if isa(v, Const)
-        if !has_free_typevars(t) && !isa(v.val, t)
-            return Bottom
-        end
+    if isa(v, Type)
+        return typeintersect(v, t)
+    end
+    has_free_typevars(t) && return v
```

Comment:
I think this is basically an NFC, but it looks a bit funny this way: Is it really safe to rely on the type intersection if `isa(v, Type)` even if `has_free_typevars(t)`? And if so, why is it not safe if `v` is something else (then using `widev = widenconst(v)` of course).

Reply:
We need to stop them from appearing, but that's a project for later. Mostly likely it would actually over-estimate the intersection (which is generally what we do here too), so we only try to avoid them most of the time.